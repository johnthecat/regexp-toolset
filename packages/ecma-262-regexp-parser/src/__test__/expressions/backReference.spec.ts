import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

describe('Back Reference', () => {
  it('should parse back reference', () => {
    expect(parseRegexp(/(a)\1/)).toMatchSnapshot();
  });

  it('should be simple escaped char without group', () => {
    expect(parseRegexp(/a\1/)).toMatchSnapshot();
  });
});
