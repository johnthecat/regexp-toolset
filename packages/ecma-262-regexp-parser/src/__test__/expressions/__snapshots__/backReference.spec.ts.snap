// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Back Reference > should be simple escaped char without group 1`] = `
{
  "body": {
    "end": 3,
    "expressions": [
      {
        "charCode": 97,
        "end": 1,
        "kind": "Char",
        "start": 1,
        "type": "Simple",
        "value": "a",
      },
      {
        "charCode": 49,
        "end": 3,
        "kind": "Char",
        "start": 2,
        "type": "Escaped",
        "value": "1",
      },
    ],
    "kind": "Alternative",
    "start": 1,
  },
  "end": 4,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Back Reference > should parse back reference 1`] = `
{
  "body": {
    "end": 5,
    "group": {
      "body": {
        "charCode": 97,
        "end": 2,
        "kind": "Char",
        "start": 2,
        "type": "Simple",
        "value": "a",
      },
      "end": 3,
      "kind": "Group",
      "specifier": null,
      "start": 1,
      "type": "capturing",
    },
    "kind": "BackReference",
    "start": 1,
  },
  "end": 6,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;
